pkg/registry/generic/matcher.go

```diff
@@ -26,6 +26,10 @@ import (
 // AttrFunc returns label and field sets for List or Watch to compare against, or an error.
 type AttrFunc func(obj runtime.Object) (label labels.Set, field fields.Set, err error)
 
+// IndexFieldFunc returns name of the field on which the index given by the name is build
+// and true. If the index with the given name is not supported, returns ("", false).
+type IndexFieldFunc func(indexName string) (string, bool)
```

Comment:
suggestion: completely delete this, and require index names to exactly match field names.